smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/member/MemberShapeDecodeGenerator.kt

```diff
@@ -147,6 +157,34 @@ open class MemberShapeDecodeGenerator(
     private fun default(memberShape: MemberShape): String {
         val targetShape = ctx.model.expectShape(memberShape.target)
         val defaultTrait = memberShape.getTrait<DefaultTrait>() ?: targetShape.getTrait<DefaultTrait>()
+        val requiredTrait = memberShape.getTrait<RequiredTrait>()
```

Comment:
do we need required trait for default trait? We shouldn't be resolving required trait ourselves I think and leave that to the knowledge index to resolve then we treat accordingly. but that's once we're ready to say if something should be required we're gonna make it non-nullable

Reply:
This here is just for putting zeros or zero-equivalent values to deserialized output members if:
- It's required
- It does not have a default trait
- AND the response from server doesn't have any value for it (wrong behavior from server)

I think we can add this implementation regardless of what we do re: optionality down the line.